string-manipulation-3/src/title-case.js

```diff
@@ -0,0 +1,162 @@
+/* exported titleCase */
+
+/** function capitlizeHyphen (hyphenWord) { //"in-depth"
+  const separatedWords = hyphenWord.split('-'); //['in', 'depth']
+  separatedWords[1] = separatedWords[1][0].toUpperCase() + separatedWords[1].slice(1); //['in', 'Depth']
+  return separatedWords.join('-'); //'in-Depth'
+} */
+
+function titleCase(title) {
+  const exceptions = ['a', 'an', 'the', 'as', 'at', 'by', 'for', 'in', 'of', 'on', 'per', 'to', 'and'];
+  const updatedTitle = [];
+  let str;
+  let specialCharPosition;
+  let specialChar;
+  if (title.includes(':')) {
+    specialCharPosition = title.indexOf(':');
+    specialChar = ':';
+    str = title.toLowerCase().split(':');
+    console.log('colon', str);
+  } else if (title.includes('-')) {
+    str = title.toLowerCase().split('-');
+  }
+  str = title.toLowerCase().split(' ');
+  console.log('wordSplit');
+  console.log(str);
+
+  for (let i = 0; i < str.length; i++) {
+    updatedTitle[i] = str[i][0].toUpperCase() + str[i].slice(1).toLowerCase();
+
+    if (exceptions.includes(str[i]) && i !== 0) {
+      updatedTitle[i] = str[i].toLowerCase();
+    } else if (str[i] === 'api') {
+      updatedTitle[i] = 'API';
+    } else if (str[i].includes('javascript')) {
+      updatedTitle[i] = 'JavaScript';
```

Comment:
based on what you have so far, you could use the `replace` method instead of just assigning the string `'JavaScript'` because doing so will git rid of the colon. It would look like this:

`updatedTitle[i] =  str[i].replace('javascript', 'JavaScript');`

Using the `replace` method will keep the colon in place in the string